lxd/instance/drivers/driver_qemu.go

```diff
@@ -2044,29 +2044,34 @@ func (d *qemu) setupNvram() error {
 }
 
 func (d *qemu) qemuArchConfig(arch int) (path string, bus string, err error) {
+	basePath := ""
+	if os.Getenv("LXD_EXT_QEMU_SNAP") != "" {
+		basePath = filepath.Join(os.Getenv("SNAP"), "opt", "external-qemu", "bin") + "/"
```

Comment:
Since we are only ever going to support Linux and there's already a "/" added, maybe we can go with the simpler:

```suggestion
		basePath = filepath.Join(os.Getenv("SNAP"), "opt/external-qemu/bin/")
```

If we envision having other external tools, maybe we could use `/snap/lxd/$REV/external/qemu`? (I'm not fan of the `/opt` bit ;) )

Reply:
Regarding reusing an existing directory, maybe we could have `lxd-pkg-snap` create an empty `external` directory at its root.

Reply:
I have fixed the code in accordance with your suggestion. Thanks, Simon!

About, a switch from the `opt` to the `external` directory. Let's ask what @tomponline thinks about it.

I personally can't see any difference between having `$SNAP/opt/external-qemu/` and `$SNAP/external/qemu/`.
But if you insist, I'm ready to rework this.

Reply:
From LXD's perspective it should just be told the path via the env var from snap.

As for how we mount the external qemu snap inside the lxd snap using the content interfaces, I don't suppose it makes a lot of difference.

The "$SNAP/external" would seem to be a reasonable place to mount content coming from external snaps, so `$SNAP/external/qemu/` seems fine to me. As long as LXD and the snap packaging are not tightly coupled we can be free to change it later if needed.